Migrate to servant v0.7

Arguggi opened the pull request:
Still TODO:

- [x] Update tests
- [x] Remove all `Token`s and switch to adding a [`baseUrlPath`](http://haddock.stackage.org/nightly-2016-05-08/servant-client-0.6.1/Servant-Common-BaseUrl.html) to `telegramBaseUrl` ? 
- [x] Add a warning in the `README.md` to use at least `nightly-2016-05-08` and add `tls-1.3.7` in the `stack.yaml` `extra-deps` of any library that uses `haskell-telegram-api` (see my [comment](https://github.com/klappvisor/haskell-telegram-api/issues/21#issuecomment-218498489))  ?

klappvisor commented:
@Arguggi Originally I thought to keep same interface for methods. Telegram URL is always the same and there is no point to pass it always, but sometimes it would be convenient to change it for tests. Same is applicable for `Manager`.

What do you think about having methods for example?
```haskell
getMe' :: Token -> Manager -> BaseUrl -> IO (Either ServantError GetMeResponse)

getMe :: Token -> IO (Either ServantError GetMeResponse) -- calls getMe'
```

Arguggi commented:
### Manager
I left the Manager explicit because in the [newManager documentation](https://hackage.haskell.org/package/http-client-0.4.28/docs/Network-HTTP-Client.html#v:newManager) it says:

>  Creating a new Manager is a relatively expensive operation, you are advised to share a single Manager between requests instead.

So we could simplify the `getMe` function and remove the `Manager` but that would mean we would have to create a new `Manager` for every api call which is probably not a good idea.

I don't have any benchmarks but I could add some if we want to see how "expensive" `newManager` actually is.

### BaseUrl

The `BaseUrl` can be removed in the simpler function version. But I was thinking that maybe we could move the `Token` to `BaseUrl` instead of repeating it in every api.

So the 2 versions would be:
````
getMeA :: Token -> Manager -> IO (Either ServantError GetMeResponse) -- Simplified function that call the original getMe
```

or 

```
getMeB :: Manager -> BaseUrl -> IO (Either ServantError GetMeResponse)  -- Remove Token from all the APIs
```

I slightly prefer the `getMeB` version since it removes the Token "boilerplate" while keeping lower the number of functions in `API.hs`. Tell me what you prefer.

klappvisor commented on test/MainSpec.hs:
```diff
@@ -25,70 +23,71 @@ nosuccess e = e `shouldSatisfy` isLeft
+    --it "should send photo" $ do
+    --  Right MessageResponse { message_result = Message { caption = Just cpt } } <-
+    --    sendPhoto token (SendPhotoRequest chatId "AgADBAADv6cxGybVMgABtZ_EOpBSdxYD5xwZAAQ4ElUVMAsbbBqFAAIC" (Just "photo caption") Nothing Nothing) manager telegramBaseUrl
+    --  cpt `shouldBe` "photo caption"
```
Why did you have to comment out that test? it's private image, isn't it? x)

Arguggi replied:
It was failing when I ran it with my bots token.

klappvisor replied:
ok, I will take a look

klappvisor commented:
@Arguggi You are right about manager... Documentation is clear, benchmarks are not needed.

I would keep `Token` as an option, but you made your point regarding `BaseUrl`. Could you please remove it from api but keep definition (line 54)? After merge I will think of its destiny. 

I would also move `Manager` and `BaseUrl` in front of request objects, but we also can merge like this and I change it later.

Arguggi commented:
I removed `BaseUrl` from all the exported functions, redifined `run` and fixed the tests. 

(BTW are you aware that all the haskell files in `src/Web/Telegram/API/Bot/` have CRLF line terminators while the test files in `test` don't?)

klappvisor commented:
oops... thanks for letting me know! I'll fix it

klappvisor commented:
Looks good to me. I don't think we need to update readme. Anyway I would not release code with nightly resolver. Hopefully we will have stable build when work on bot-2.0 api is over...

Arguggi commented:
I rebased my bot-2.0 branch on your latest master. I'll close this and open a new PR. I also fixed [issue 20](https://github.com/klappvisor/haskell-telegram-api/issues/20) on my bot-2.0 branch.

klappvisor commented:
@Arguggi Good job! Thank you!
